### PR TITLE
IGNITE-22800 Make field BplusTree#globalRmvId monotonically increasing to avoid errors after node recovery

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-22800

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/AbstractPageMemoryTableStorage.java [hunk 1]
```diff
@@ -93,6 +93,9 @@ public StorageIndexDescriptorSupplier getIndexDescriptorSupplier() {
      */
     public abstract DataRegion<?> dataRegion();
 
+    /** Returns storage engine for the table. */
+    public abstract AbstractPageMemoryStorageEngine engine();
```
Comment (ibessonov) [inline modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/AbstractPageMemoryTableStorage.java:97]: `protected` maybe? Why would it be public.
Whatever you decide

File: modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/mv/PersistentPageMemoryMvPartitionStorageConcurrencyTest.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ void setUp(
             @InjectConfiguration("mock.checkpoint.checkpointDelayMillis = 0")
             PersistentPageMemoryStorageEngineConfiguration engineConfig,
             @InjectConfiguration("mock.profiles.default = {engine = \"aipersist\"}")
-            StorageConfiguration storageConfiguration
+            StorageConfiguration storageConfig
```
Comment (ibessonov) [inline modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/mv/PersistentPageMemoryMvPartitionStorageConcurrencyTest.java:55]: It's strange that you've shortened this name. What motivated you?

**Review by ibessonov — APPROVED** [review]
Looks good to me. Too many unnecessary renames, but I don't want to argue about that anymore

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/AbstractPageMemoryTableStorage.java (hunk 1 above)
Comment (tkalkirill) [reply modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/AbstractPageMemoryTableStorage.java:97]: This method is used for example in package `org.apache.ignite.internal.storage.pagememory.mv`.

File: modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/mv/PersistentPageMemoryMvPartitionStorageConcurrencyTest.java (hunk 2 above)
Comment (tkalkirill) [reply modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/mv/PersistentPageMemoryMvPartitionStorageConcurrencyTest.java:55]: For consistency in fields configuration naming, previously there was only `engineConfig`, then they brought `storageConfiguration`, it confused me a little, I can do it in a separate task.

Comment (ibessonov) [reply modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/mv/PersistentPageMemoryMvPartitionStorageConcurrencyTest.java:55]: I'm fine with it being here

**Pull request merged**